pkg/upgrade/cache.go

```diff
@@ -43,11 +43,11 @@ type DownloaderOption interface {
 	apply(d *Downloader)
```

Comment:
it's not done like that, please make sure to remove the interface, 

you can have a new type e.g DownloaderOption that's a mutation method func(d  *Downloader)

and create function e.g WtihNumWorkers e.g

```go
func WithNumWorkers(n int) func(*Downloader) {
            return func(d *Downloader){ 
                    d.numWorkers = n
            
            }
}

```


Please don't introduce interfaces that're not needed

Reply:
we use that in order not to break the API, while this is all internal it's fine to have the numworker in the factory function, or even if the type is simple an depends on the zero value you can directly use it and have the fields exported without tons of chores.